### PR TITLE
[Merged by Bors] - feat: more API for the right homology of short complexes

### PR DESCRIPTION
**Pull request by joelriou**
This PR develops more API for the right homology of short complexes. All definitions and statements are parallel to their left homology counterparts.

- [x] depends on: #6008 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean [hunk 1]
```diff
@@ -721,6 +721,177 @@ lemma opcyclesMap_comp [HasRightHomology S₁] [HasRightHomology S₂] [HasRight
...
+  inv := rightHomologyMap e.inv
+  hom_inv_id := by rw [← rightHomologyMap_comp, e.hom_inv_id, rightHomologyMap_id]
+  inv_hom_id := by rw [← rightHomologyMap_comp, e.inv_hom_id, rightHomologyMap_id]
+
+instance isIso_rightHomologyMap_of_iso (φ : S₁ ⟶ S₂) [IsIso φ] [S₁.HasRightHomology]
+    [S₂.HasRightHomology] :
+    IsIso (rightHomologyMap φ) :=
+  (inferInstance : IsIso (rightHomologyMapIso (asIso φ)).hom)
+
+/-- The isomorphism `S₁.cycles ≅ S₂.cycles` induced by an isomorphism
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean:769]:
```suggestion
/-- The isomorphism `S₁.opcycles ≅ S₂.opcycles` induced by an isomorphism
```

File: Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean [hunk 2]
```diff
@@ -721,6 +721,177 @@ lemma opcyclesMap_comp [HasRightHomology S₁] [HasRightHomology S₂] [HasRight
...
+
+variable (C)
+variable [HasKernels C] [HasCokernels C]
+
+/-- The right homology functor `ShortComplex C ⥤ C`, where the right homology of a
+short complex `S` is understood as a kernel of the obvious map `S.fromOpcycles : S.opcycles ⟶ S.X₃`
+where `S.opcycles` is a cokernel of `S.f : S.X₁ ⟶ S.X₂`. -/
+@[simps]
+noncomputable def rightHomologyFunctor :
+    ShortComplex C ⥤ C where
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean:861]: In the corresponding development of left homology, this is formatted as
```suggestion
noncomputable def rightHomologyFunctor : ShortComplex C ⥤ C where
```
It's fine to leave it like it is, of course, just something I noticed.

**Review by TwoFX — APPROVED** [review]
Thanks!
bors d+

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**